ghost/core/core/server/api/endpoints/utils/serializers/output/utils/extra-attrs.js

```diff
@@ -8,31 +8,35 @@ const readingMinutes = require('@tryghost/helpers').utils.readingMinutes;
  * @returns {void} - modifies attrs
  */
 module.exports.forPost = (options, model, attrs) => {
-    // This function is split up in 3 conditions for 3 different purposes:
+    const columnsIncludesCustomExcerpt = options.columns?.includes('custom_excerpt');
+    const columnsIncludesExcerpt = options.columns?.includes('excerpt');
+    const columnsIncludesPlaintext = options.columns?.includes('plaintext');
+    const columnsIncludesReadingTime = options.columns?.includes('reading_time');
+    const formatsIncludesPlaintext = options.formats?.includes('plaintext');
+
     // 1. Gets excerpt from post's plaintext. If custom_excerpt exists, it overrides the excerpt but the key remains excerpt.
-    if (Object.prototype.hasOwnProperty.call(options, 'columns') || options.columns?.includes('excerpt') || (options.columns?.includes('excerpt') && options.formats?.includes('plaintext'))) {
-        if (options.columns?.includes('excerpt')) {
-            if (!attrs.custom_excerpt || attrs.custom_excerpt === null) {
-                let plaintext = model.get('plaintext');
-                if (plaintext) {
-                    attrs.excerpt = plaintext.substring(0, 500);
-                } else {
-                    attrs.excerpt = null;
-                }
-                if (!options.columns.includes('custom_excerpt')) {
-                    delete attrs.custom_excerpt;
-                }
+    if (columnsIncludesExcerpt || (columnsIncludesExcerpt && formatsIncludesPlaintext)) {
+        if (!attrs.custom_excerpt) {
+            let plaintext = model.get('plaintext');
+            if (plaintext) {
+                attrs.excerpt = plaintext.substring(0, 500);
             } else {
-                attrs.excerpt = attrs.custom_excerpt;
-                if (!options.columns?.includes('custom_excerpt')) {
-                    delete attrs.custom_excerpt;
-                }
+                attrs.excerpt = null;
+            }
+        } else {
+            attrs.excerpt = attrs.custom_excerpt;
+            if (!columnsIncludesCustomExcerpt) {
+                delete attrs.custom_excerpt;
             }
```

Comment:
Looks like this is an unnecessary duplicate as it's already run for either condition later 🙂

Reply:
Lol, the AI rabbit got there too. I'll be quiet for now 🤐

Reply:
My new best friend @coderabbitai also made this suggestion 😀